### PR TITLE
Adds management command for creating users from google gsheet

### PR DESCRIPTION
**Pull request by esirK**
## Description

Pulls users from a google sheet and creates them and adds them to the database.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: gmmp/management/commands/create_users.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        for record in records:
+            # Create users
+            email = record.get('Email')
+            first_name = record.get('Firstname')
+            last_name = record.get('Lastname')
+            username = record.get('Username')
+            country = record.get('Country')
+            group, _ = Group.objects.get_or_create(name=record.get('Designation'))
+            user, _ = User.objects.get_or_create(email=email, first_name=first_name, last_name=last_name, username=username)
+            user.set_password("RandomPassword321")
```
Comment (kilemensi) [inline gmmp/management/commands/create_users.py]: This should be read from env `DEFAULT_PASSWORD` or something similar.

File: gmmp/management/commands/create_users.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        records = worksheet.get_all_records()
+
+        for record in records:
+            # Create users
+            email = record.get('Email')
+            first_name = record.get('Firstname')
+            last_name = record.get('Lastname')
+            username = record.get('Username')
+            country = record.get('Country')
+            group, _ = Group.objects.get_or_create(name=record.get('Designation'))
```
Comment (kilemensi) [inline gmmp/management/commands/create_users.py]: How are we assigning permissions to these group i.e. manually or is it possible to also do it in a script?

**Review by kilemensi — COMMENTED** [review]
Not sure @esirK 

--------------------

**Problems**

1. We now have **2** ways of handling gSheets in the same app,
2. Each of these ways of accessing gSheets, load credentials from different env variables (not sure if they have same scope or not).

We need to _harmonize_ this.

**Questions**

1. Looks like we're not clearing existing users. How do make sure about unique stuff e.g. if same user but now they have different email or username, how do we make sure their old account is deactivated? I'm thinking there is a need to clear out all users/groups before adding new ones from the sheet.
2. _Not 💯  sure_ but the **app** is still using 2 country code by default (django-country defaults to alpha2)... I don't see us translating the gSheets alpha3 to alpha2 when creating country object.

**Missing**

Lastly we also need to **clear all the data** created by existing users that is already in the system... If there is a foreign key relation with cascade, then deleting existing users should delete data as well. Otherwise, we need to delete it _manually_.

File: gmmp/management/commands/create_users.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        SHEET_KEY = os.environ.get("SHEET_KEY")
+        GOOGLE_CREDENTIALS = json.loads(os.environ.get("GOOGLE_CREDENTIALS", '{}'))
+        scope = ["https://spreadsheets.google.com/feeds", "https://www.googleapis.com/auth/drive",]
+
+        credentials = ServiceAccountCredentials.from_json_keyfile_dict(GOOGLE_CREDENTIALS, scopes=scope)
+
+        client = gspread.authorize(credentials)
+
+        sheet = client.open_by_key(SHEET_KEY)
+        worksheet = sheet.worksheet('data')
```
Comment (kilemensi) [inline gmmp/management/commands/create_users.py]:
```suggestion
        worksheet = sheet.worksheet('DATA')
```

File: gmmp/management/commands/create_users.py (hunk 2 above)
Comment (esirK) [reply gmmp/management/commands/create_users.py]: right now we have to assign them manually. Will look into how to do it using code.

**Pull request merged**